test_runner/regress/test_tenant_detach.py

```diff
@@ -78,9 +79,29 @@ def test_tenant_reattach(
         ".*failed to perform remote task UploadMetadata.*, will retry.*"
     )
 
+    ps_metrics = parse_metrics(pageserver_http.get_metrics(), "pageserver")
+    tenant_metric_filter = {
+        "tenant_id": str(tenant_id),
+        "timeline_id": str(timeline_id),
+    }
+    pageserver_last_record_lsn_before_detach = int(
+        ps_metrics.query_one("pageserver_last_record_lsn", filter=tenant_metric_filter).value
+    )
+
```

Comment:
I was just let know of the `wait_for_upload` yesterday, why is not needed here? Is it luck?

Because the added assert on line 103 is not always true, or perhaps it could be read from remote_consistent_lsn on line 88 as well.

Reply:
On an unrelated note: the postgres is kept running while doing tenant detach and attach. I am surprised that this works.

On line 63 there could also be:

```patch
- pg = env.postgres.create_start("main", tenant_id)
+ with env.postgres.create_start("main", tenant_id) as pg:
```

Which would have to be repeated, probably with `start` if we don't want to test that postgres will reconnect in this test.

Reply:
wait_for_upload is already in line 72